src/c/psychrolib.c

```diff
@@ -286,6 +290,41 @@ double GetRelHumFromVapPres     // (o) Relative humidity [0-1]
   return VapPres/GetSatVapPres(TDryBulb);
 }
 
+// Helper function returning the derivative of the natural log of the saturation vapor pressure
+// as a function of dry-bulb temperature.
+// Reference: ASHRAE Handbook - Fundamentals (2017) ch. 1 eqn. 5 & 6
+double dLnPws_        // (o) Derivative of natural log of vapor pressure of saturated air in Psi [IP] or Pa [SI]
+  ( double TDryBulb   // (i) Dry bulb temperature in °F [IP] or °C [SI]
+  )
+{
+  double dLnPws, T;
+
+  if (isIP())
+  {
+    T = GetTRankineFromTFahrenheit(TDryBulb);
+
+    if (TDryBulb >= -148. && TDryBulb <= 32.)
```

Comment:
In C and JS you test TDryBulb against the freezing point but also against the domain of validity of the expression. I prefer the approach you take in the other languages where you test only against the freezing point. When we reach this helper function we have already checked that TDryBulb is within the domain of validity of the expression so there is no need to test it again. Furthermore, the implementation proposed here does not cover the case where TDryBulb would fall outside the domain of validity.